HealthcareApp/src/Healthcare.Infrastructure/Repository/AppointmentRepository.cs

```diff
@@ -54,4 +54,10 @@ public async Task<bool> IsAvailableAsync(string doctorId, DateTime date, TimeSpa
 
         return existingAppointment == null;
     }
+
+    public async Task MigrateRangeAsync(List<Appointment> appointments)
+    {
+        await _appDbContext.Appointments.AddRangeAsync(appointments);
+        await _appDbContext.SaveChangesAsync();
```

Comment:
You work with too many data in memory. Think how to avoid it and fix

Reply:
i probably did not understand it, you will see in the next commit that I left the similar segment of code in new MigrationsRepository file. maybe you could give me some tip how here I can refine the performance

Reply:
I propose you to use batching here. Because there could be millions of appointments, and writing them in a single query could be a pressure to DB and will lock Appointments table for too long, so that your application will have DB lag